source/Octopus.Server.Client/Operations/RegisterMachineOperation.cs

```diff
@@ -303,32 +284,10 @@ async Task<List<EnvironmentResource>> GetEnvironments(IOctopusSpaceAsyncReposito
 
             if (Environments is not null && Environments.Any())
             {
-                var envsByName = await repository.Environments.FindByNames(Environments).ConfigureAwait(false);
-                environments.AddRange(envsByName);
-
-                var missing = Environments
-                    .Except(envsByName.Select(e => e.Name), StringComparer.OrdinalIgnoreCase)
-                    .ToArray();
-
-                //use the missing names to try and find by slug
-                var environmentsBySlug = await repository.Environments.FindBySlugs(missing, CancellationToken.None)
-                    .ConfigureAwait(false);
-                environments.AddRange(environmentsBySlug);
-
-                missing = missing
-                    .Except(environmentsBySlug.Select(e => e.Slug), StringComparer.OrdinalIgnoreCase)
-                    .ToArray();
-
-                //any other missing slugs/names could be Id's, so looks again
-                var environmentByIds = await repository.Environments.Get(missing).ConfigureAwait(false);
-                environments.AddRange(environmentByIds);
-
-                missing = missing
-                    .Except(environmentByIds.Select(e => e.Id), StringComparer.OrdinalIgnoreCase)
-                    .ToArray();
-
-                if (missing.Any())
-                    throw new InvalidRegistrationArgumentsException(CouldNotFindByMultipleMessage("environment", missing.ToArray()));
+                var environmentsByNameIdOrSlug =
```

Comment:
Yea either way is fine really 🤷 It also doesn't hurt to be explicit, especially because the chunk of code above it is also adding environments that are found by name to the resulting env list.